arcade/examples/sprite_rotate_tank.py

```diff
@@ -0,0 +1,185 @@
+                                                     center_x=x * 64, center_y=y * 64))
+
+        # The tank and barrel sprite. These Can be all added to their own separate class in a more advanced program.
+        self.tank = arcade.Sprite(":resources:images/topdown_tanks/tankBody_dark_outline.png",
+                                  center_x=SCREEN_WIDTH // 2, center_y=SCREEN_HEIGHT // 2)
+        self.barrel = RotatingSprite(":resources:images/topdown_tanks/tankDark_barrel3_outline.png",
+                                     center_x=SCREEN_WIDTH // 2, center_y=SCREEN_HEIGHT // 2 - 13)
+
+        self.tank_direction = 0.0  # If the tank is moving forward or backwards.
+        self.tank_turning = 0.0  # If the tank is turning left or right.
+
+        self.mouse_pos = [0, 0]
+
+        self.tank_sprite_list = arcade.SpriteList()
+        self.tank_sprite_list.extend([self.tank, self.barrel])
+
+        self._correct = True
+        self.correct_text = arcade.Text("Turret Rotation is Correct, Press P to Switch",
+                                        SCREEN_WIDTH // 2, SCREEN_HEIGHT - 25, anchor_x='center')
+
+        self.control_text = arcade.Text("WASD to move tank, Mouse to aim", SCREEN_WIDTH // 2, 15, anchor_x='center')
+
+    def on_draw(self):
+        self.clear()
+        self.background.draw()
+        self.tank_sprite_list.draw()
+
+        self.control_text.draw()
+        self.correct_text.draw()
+
+    def on_update(self, delta_time: float):
+        self.move_tank(delta_time)
+
+    def move_tank(self, delta_time):
+        """
+        Perform all calculations about how to move the tank, including both the body and the barrel
+        """
+
+        # update the angle of the tank's body alone. The barrel will be updated after the body is moved
+        self.tank.angle += TANK_SPEED * self.tank_turning * delta_time
+
+        # find how much the tank's x and y should change to move forward or back.
+        x_dir = math.cos(self.tank.radians - math.pi / 2) * self.tank_direction * TANK_SPEED * delta_time
+        y_dir = math.sin(self.tank.radians - math.pi / 2) * self.tank_direction * TANK_SPEED * delta_time
+
+        # we then move the tank and the barrel since they are connected together.
+        self.tank.center_x += x_dir
+        self.tank.center_y += y_dir
+
+        self.barrel.center_x += x_dir
+        self.barrel.center_y += y_dir
+
+        if self.correct:
+            # Rotate the barrel sprite around the center of the tank, not the center of the barrel sprite
+
+            # we need to add 90 to the angle due to orientation of the barrel texture.
+            # we need to remove the barrels angle as we only want the change in angle.
+            angle_change = arcade.get_angle_degrees(self.tank.center_y, self.tank.center_x,
+                                                    self.mouse_pos[1], self.mouse_pos[0]) - self.barrel.angle + 90
```

Comment:
You could try something like this to reduce the horizontal space taken up by the code:
```py
angle_change = arcade.get_angle_degrees(
    self.tank.center_y, self.tank.center_x,
    self.mouse_pos[1], self.mouse_pos[0])
    
# subtract the current angle because we want only the change in angle to rotate the barrel
angle_change -= self.barrel.angle

# compensate for the vertical orientation of the sprite's image file
angle_change += 90
```